src/remoteConnector.ts

```diff
@@ -580,31 +580,12 @@ export class RemoteConnector extends Disposable {
 		});
 	}
 
-	private async updateRemoteSSHConfig(usingSSHGateway: boolean, localAppSSHConfigPath: string | undefined) {
-		const remoteSSHconfig = vscode.workspace.getConfiguration('remote.SSH');
-		const defaultExtConfigInfo = remoteSSHconfig.inspect<string[]>('defaultExtensions');
-		const defaultExtensions = defaultExtConfigInfo?.globalValue ?? [];
-		if (!defaultExtensions.includes('gitpod.gitpod-remote-ssh')) {
-			defaultExtensions.unshift('gitpod.gitpod-remote-ssh');
-			await remoteSSHconfig.update('defaultExtensions', defaultExtensions, vscode.ConfigurationTarget.Global);
-		}
-
-		const currentConfigFile = remoteSSHconfig.get<string>('configFile');
-		if (usingSSHGateway) {
-			if (currentConfigFile?.includes('gitpod_ssh_config')) {
-				await remoteSSHconfig.update('configFile', undefined, vscode.ConfigurationTarget.Global);
-			}
-		} else {
-			// TODO(ak) notify a user about config file changes?
-			if (currentConfigFile === localAppSSHConfigPath) {
-				// invalidate cached SSH targets from the current config file
-				await remoteSSHconfig.update('configFile', undefined, vscode.ConfigurationTarget.Global);
-			}
-			await remoteSSHconfig.update('configFile', localAppSSHConfigPath, vscode.ConfigurationTarget.Global);
+	private async ensureRemoteSSHExtInstalled(flow: UserFlowTelemetryProperties): Promise<boolean> {
+		const isOfficialVscode = vscode.env.uriScheme === 'vscode' || vscode.env.uriScheme === 'vscode-insiders';
```

Comment:
Should not we ensure that yours extension is installed then? Should we also collect your extension version then too as part of error reporting/tracing/analytics?

Reply:
it's tricky because cursor forked my extension so it has another id 🤦

Reply:
I mean we can have a case if it is not cursor then?

But the best would be that Cursor folks contribute back to your extension and don't cause discrepancy cc @mbrevoort  if you know whom to cc :)